test/Locker.test.ts

```diff
@@ -0,0 +1,56 @@
+import { Contract } from "ethers";
+import { expect } from "./chai-setup";
+import { pack, keccak256 } from "@ethersproject/solidity";
+import { ethers } from "hardhat";
+import {
+  setupNftGemGovernor
+} from './fixtures/Governance.fixture';
+
+describe('Locker Test Suite', function () {
+  let locker: Contract;
+  let NFTGemMultiToken: any;
+  const tokenHash = keccak256(['bytes'], [pack(['string'], ['Test Token'])]);
+  beforeEach(async () => {
+    // Deploy Locker Contract on each test iteration
+    locker = await (await ethers.getContractFactory('Locker')).deploy();
+    const setupNftGemGovernorResult = await setupNftGemGovernor();
+    NFTGemMultiToken = setupNftGemGovernorResult.NFTGemMultiToken;
+    const [sender] = await ethers.getSigners();
+    // Mint token to sender to make sure the sender have enough balance
+    await NFTGemMultiToken.mint(sender.address, tokenHash, 100);
+    await NFTGemMultiToken.connect(sender).setApprovalForAll(locker.address, true);
+  })
+  it('Drop Off', async function () {
```

Comment:
Why are you picking an edge case for pick up and drop off? You are using the same token hash for both your unlock token and your award token. While this is in theory possible it's not the primary case you should be testing.  Instead, above in your beforeEach, create 2 different token hashes. They can just be 1 and 2 for ease of the test. 1 is just as good as maxuint256 in the test and easier to read and inderstand in a test case.